adapt/intent.py

```diff
@@ -130,6 +130,7 @@ def __init__(self, name, requires, at_least_one, optional):
         self.requires = requires
         self.at_least_one = at_least_one
         self.optional = optional
+        self.excludes = excludes or []
```

Comment:
won't code in the wild break left and right if we introduce this new arg as being mandatory?
a new arg will make migration to the latest version painful, something worth keeping in mind

consistency is also important and this will now bug me forever :P

Reply:
That's a fair point; I had not considered folks that use the Intent initializer directly; I imagine most usage is via the builder, but can't guarantee it.